languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10026,6 +10026,35 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="QUELLE_OU_QU_ELLE" name="quelle ou qu'elle" default="temp_off">
+          <rule>
+              <pattern>
+                  <token postag="SENT_START" postag_regexp="yes"></token>
+                  <marker>
+                      <token regexp="yes">Qu(')?</token>
```

Comment:
Per se, `qu'?` would suffice (the question mark makes only the `'` optional). No need to put an uppercase `Q` in there. Unless your token is `case_sensitive="yes"`, both `q` and `Q` will be matched anyway. (It wouldn't make sense to make this token case sensitive, by the way).

You could use the apostrophe entity here: `<token regexp="yes">qu(&apostrophe;)?</token>`

With the `?` you put, you make the apostrophe optional, the rule will also catch "Qu elle est ta question ?" (just thought I'd mention it)